### ISSUE TITLE
Nested generic types result in "Unbound type parameter, missing type node."

### ISSUE THREAD

**Issue by geomaster** [woutervh-/typescript-is#4]
Not sure if this is intended behavior, please let me know if so!

**ttypescript version**: 3.2.4
**typescript-is version**: 0.7.17
**node version**: 11.6.0

**Steps to reproduce**:
Try compiling a file containing the following:
```typescript
import { is } from 'typescript-is';

interface Wrapped<T> {
    wrap: T;
}

interface X<T> {
    a: Wrapped<T>;
}

is<X<number>>({ a: { wrap: 2 } });
```

**Actual results**:
The compilation fails with an error like this:
```
typescript-is: transforming program with 30 source files; using TypeScript 3.2.4.

undefined:86995
                throw e;
                ^
NestedError: Failed to transform node at: /home/geomaster/Projects/tests/typescript-is/index.ts:9:2
    at transformNodeAndChildren (/home/geomaster/Projects/tests/typescript-is/node_modules/typescript-is/lib/transform-inline/transformer.js:27:15)
    at ts.visitEachChild (/home/geomaster/Projects/tests/typescript-is/node_modules/typescript-is/lib/transform-inline/transformer.js:29:62)
    at visitNode (/home/geomaster/Projects/tests/typescript-is/node_modules/typescript/lib/typescript.js:63390:23)
    at Object.visitEachChild (/home/geomaster/Projects/tests/typescript-is/node_modules/typescript/lib/typescript.js:63664:59)
    at transformNodeAndChildren (/home/geomaster/Projects/tests/typescript-is/node_modules/typescript-is/lib/transform-inline/transformer.js:29:15)
    at ts.visitEachChild (/home/geomaster/Projects/tests/typescript-is/node_modules/typescript-is/lib/transform-inline/transformer.js:29:62)
    at visitNodes (/home/geomaster/Projects/tests/typescript-is/node_modules/typescript/lib/typescript.js:63441:48)
    at visitLexicalEnvironment (/home/geomaster/Projects/tests/typescript-is/node_modules/typescript/lib/typescript.js:63474:22)
    at Object.visitEachChild (/home/geomaster/Projects/tests/typescript-is/node_modules/typescript/lib/typescript.js:63778:54)
    at transformNodeAndChildren (/home/geomaster/Projects/tests/typescript-is/node_modules/typescript-is/lib/transform-inline/transformer.js:29:15)
Caused By: Error: Unbound type parameter, missing type node.
    at visitTypeParameter (/home/geomaster/Projects/tests/typescript-is/node_modules/typescript-is/lib/transform-inline/visitor.js:255:15)
    at visitType (/home/geomaster/Projects/tests/typescript-is/node_modules/typescript-is/lib/transform-inline/visitor.js:362:16)
    at visitTypeParameter (/home/geomaster/Projects/tests/typescript-is/node_modules/typescript-is/lib/transform-inline/visitor.js:257:12)
    at visitType (/home/geomaster/Projects/tests/typescript-is/node_modules/typescript-is/lib/transform-inline/visitor.js:362:16)
    at createPropertyCheck (/home/geomaster/Projects/tests/typescript-is/node_modules/typescript-is/lib/transform-inline/visitor.js:8:28)
    at visitPropertySignature (/home/geomaster/Projects/tests/typescript-is/node_modules/typescript-is/lib/transform-inline/visitor.js:41:16)
    at visitDeclaration (/home/geomaster/Projects/tests/typescript-is/node_modules/typescript-is/lib/transform-inline/visitor.js:49:16)
    at visitPropertySymbol (/home/geomaster/Projects/tests/typescript-is/node_modules/typescript-is/lib/transform-inline/visitor.js:77:25)
    at visitRegularObjectType (/home/geomaster/Projects/tests/typescript-is/node_modules/typescript-is/lib/transform-inline/visitor.js:131:29)
    at visitObjectType (/home/geomaster/Projects/tests/typescript-is/node_modules/typescript-is/lib/transform-inline/visitor.js:172:16)
```

**Expected results**:
Since the `T` type parameter is bound to `number` in `X<number>`, I'd expect `wrap` to also be of known type, precisely, `Wrapped<number>`. However, it looks like typescript-is isn't able to deduce this, and so thinks that `T` in `a: Wrapped<T>` is an unbound type parameter.

Thanks and let me know if I can help with more info!

**Comment by woutervh-** [comment]
Thanks @geomaster for the report. This does indeed look like a bug. I will investigate it and hopefully deploy a fix soon!

**Comment by woutervh-** [comment]
I've made a fix, please try version 0.7.18.
You can see the test file here: https://github.com/woutervh-/typescript-is/blob/master/test/issue-4.ts

**Comment by geomaster** [comment]
Works like a charm now. Thank you so much @woutervh-!
